qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -0,0 +1,175 @@
+    it to 320 samples.
+    """
+
+    # Location where default parameter values are stored. These may be updated at construction.
+    __default_values__ = dict()
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the library.
+
+        Args:
+            default_values: A dictionary to override library default parameter values.
+        """
+        self._schedules = dict()
+
+        if default_values is not None:
+            self.__default_values__.update(default_values)
+
+    def __getitem__(self, name: str) -> ScheduleBlock:
+        """Return the schedule."""
+        if name not in self._schedules:
+            raise CalibrationError(f"Gate {name} is not contained in {self.__class__.__name__}.")
+
+        return self._schedules[name]
+
+    @property
+    def basis_gates(self) -> List[str]:
+        """Return the basis gates supported by the library."""
+        return list(name for name in self._schedules)
+
+    @abstractmethod
+    def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, ScheduleBlock]]:
+        """Return the default values for the parameters.
+
+        Returns
+            A list of tuples is returned. These tuples are structured so that instances of
+            :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
+        """
+
+
+class FixedFrequencyTransmon(BasisGateLibrary):
+    """A library of gates for fixed-frequency superconducting qubit architectures.
+
+    Note that for now this library supports single-qubit gates and will be extended
+    in the future.
+    """
+
+    __default_values__ = {"duration": 160, "amp": 0.5, "β": 0.0}
+
+    def __init__(self, default_values: Optional[Dict] = None, use_drag: bool = True):
+        """Setup the schedules.
+
+        Args:
+            default_values: Default values for the parameters this dictionary can contain
+                the following keys: "duration", "amp", "β", and "σ". If "σ" is not provided
+                this library will take one fourth of the pulse duration as default value.
+            use_drag: If set to False then Gaussian pulses will be used instead of DRAG
+                pulses.
+        """
+        super().__init__(default_values)
+
+        dur = Parameter("duration")
+        sigma = Parameter("σ")
+
+        def _beta(use_drag):
+            return Parameter("β") if use_drag else None
+
+        # X gates
+        sched_x = self._single_qubit_schedule("x", dur, Parameter("amp"), sigma, _beta(use_drag))
+
+        # Y gates
+        sched_y = self._single_qubit_schedule("y", dur, Parameter("amp"), sigma, _beta(use_drag))
+
+        # square-root X gates
+        sched_sx = self._single_qubit_schedule("sx", dur, Parameter("amp"), sigma, _beta(use_drag))
+
+        # square-root Y gates
+        sched_sy = self._single_qubit_schedule("sy", dur, Parameter("amp"), sigma, _beta(use_drag))
```

Comment:
These pulses seem to have the different parameter entry without linking. How do `*1j` or `/2` work?

Reply:
The `*1j` and `/2` work in the default values of the parameters. See here: https://github.com/eggerdj/qiskit-experiments/blob/84ee2374e1521d201f298f42716d031eb0e2d8f0/qiskit_experiments/calibration_management/basis_gate_library.py#L167-L171

 After discussing a bit with Will on the demo NB (see here: https://github.com/Qiskit/qiskit-experiments/pull/184#discussion_r673135408). I think it's better to unlink the amplitudes and DRAG parameters.

Reply:
My concern in #184 was in keeping the meaning of the parameter consistent across time. At the beginning `amp` meant the full amplitude for the `x` schedule but twice the amplitude for the `sx` schedule because it used the same `Parameter("amp")` with a factor of 1/2 in the schedule definition and then for the fine cal the `sx` schedule was redefined to unlink `sx` from `x` and changed what `amp` meant by a factor of 2.

Here I think it would be okay to link the amplitudes and betas between `x` and `y` and between `sx` and `sy` (but not between `x` and `sx`).

Reply:
Sounds good. See here: 8282ee0